18_regexp_lab/01.py

```diff
@@ -9,8 +9,9 @@
 
 with open(kuku.path, 'r') as f:
     for line in f:
-        if kuku.key in line:
-            if re.search('^\s+#', line):
-                continue
+        temp = line.split()
+        if temp[0].startswith('#'):
+            continue
+        elif temp[0] == kuku.key:
             res = re.search("=\s+(\w*)", line)
             print(res.group(1))

```

Comment:
This seems to select the proper key, but now you are just avoiding the regex. What you could have done earlier was using `*` to match any number of spaces in-front of the `#`.

```
if re.search('^\s*#', line):
   continue
```
The point if to play with regexes.

Anyway, there is still a problem. Try a prop file with
```
name = joe doe
```
Your code prints only 'joe' and leaves 'doe' out.